### PR TITLE
libc/wqueue : Block the same work comes in workqueue

### PR DESCRIPTION
**Pull request by YiHyunJin**
If same work comes in workqueue, it will not work. So the same work block.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f8239762f28e2a63fe213df5e5aedf444c6ffb1a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f8239762f28e2a63fe213df5e5aedf444c6ffb1a] - Code Rule Check Result:
lib/libc/wqueue/work_queue.c:132: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [40cd15a434022b3d2ae5627c0bed8b5d97585a72] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [40cd15a434022b3d2ae5627c0bed8b5d97585a72] - Code Rule Check Result:
lib/libc/wqueue/work_queue.c:132: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [3081416749efcf9d230f77ca3bdcbd146d1118ca] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3081416749efcf9d230f77ca3bdcbd146d1118ca] - Code Rule Check Result:
lib/libc/wqueue/work_queue.c:132: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [675ba48092206119a3d9e17b2fba132754516f7c] - Code Rule Check Result:
lib/libc/wqueue/work_queue.c:132: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [675ba48092206119a3d9e17b2fba132754516f7c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2fbbe09325ce8b6fd318323e5e5508361c3893be] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@YiHyunJin As we discussed, let's merge user workqueue into kernel's.

File: lib/libc/wqueue/work_queue.c [hunk 1]
```diff
@@ -122,21 +122,49 @@ static int work_qqueue(FAR struct usr_wqueue_s *wqueue, FAR struct work_s *work,
...
+	cur_work = (struct work_s *)wqueue->q.head;
+	while (cur_work != NULL) {
+		if ((cur_work->delay > delay) && (next_work == NULL)) {
+			next_work = cur_work;
+			break;
+		}
+		cur_work = (struct work_s *)cur_work->dq.flink;
+	}
 
+	if (next_work) {
```
Comment (jeongchanKim) [inline lib/libc/wqueue/work_queue.c]:
```
if (next_work) {
   dq_addbefore(...);
} else
#else
{
   dq_addlast(...);
}
```

File: lib/libc/wqueue/work_usrthread.c [hunk 2]
```diff
@@ -141,11 +141,17 @@ void work_process(FAR struct usr_wqueue_s *wqueue)
...
 	FAR void *arg;
 	clock_t elapsed;
-	uint32_t remaining;
 	clock_t stick;
 	clock_t ctick;
 	uint32_t next;
 	int ret;
+#ifdef CONFIG_LIB_USRWORK_SORTING
+	sigset_t set;
+	sigemptyset(&set);
```
Comment (jeongchanKim) [inline lib/libc/wqueue/work_usrthread.c]: Please add `#include <signal.h>`

File: lib/libc/wqueue/work_usrthread.c [hunk 3]
```diff
@@ -257,28 +270,40 @@ void work_process(FAR struct usr_wqueue_s *wqueue)
...
+
+				remaining = wqueue->delay - elapsed;
+				next = MIN(next, remaining);
+			}
+#endif
+				/* Wait a while to check the work list.  We will wait here until
+				* either the time elapses or until we are awakened by a signal.
+				* Interrupts will be re-enabled while we wait.
+				*/
+				usleep(next * USEC_PER_TICK);
```
Comment (an4967) [inline lib/libc/wqueue/work_usrthread.c]: This would be better to fit with other indentations.

**Pull request closed without merging**